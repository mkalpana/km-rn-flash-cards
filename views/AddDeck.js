import React, { Component } from 'react';
import { Text, StyleSheet, KeyboardAvoidingView } from 'react-native';
import { NavigationActions } from 'react-navigation'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchDecks } from '../actions';
import { Button, Input } from '../components';
import { saveDeckTitle } from '../utils/storage';
import { transparent } from '../utils/colors';

export class AddDeck extends Component{
  state = {
    title: ''
  }

  _onSaveDeck = () => {
    const { title } = this.state;
    saveDeckTitle(title)
      .then(() => {
        this.props.fetchDecks();
        const resetAction = NavigationActions.reset({
          index: 1,
          actions: [
            NavigationActions.navigate({ routeName: 'DeckList'}),
            NavigationActions.navigate({ routeName: 'Deck', params: { title }})
          ]
        })
        this.props.navigation.dispatch(resetAction);
        this.setState({ title: '' });
      });
  }

  render() {
    return (
      <KeyboardAvoidingView style={styles.container} behavior="padding">
        <Text style={styles.description}>What is the title of your new deck?</Text>
        <Input
          underlineColorAndroid={transparent}
          placeholder="Deck Title"
          value={this.state.title}
          onChangeText={(title) => this.setState({ title })}
        />
        <Button
          style={styles.btnStyle}
          label="Create Deck"
          onPress={this._onSaveDeck}
          disabled={this.state.title === ''}
        />
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20
  },
  description: {
    fontSize: 30,
    padding: 10,
    marginBottom: 10,
    alignSelf: 'center',
    textAlign: 'center'
  },
  btnStyle: {
    marginTop: 40,
    marginBottom: 10
  }
});

AddDeck.propTypes = {
  fetchDecks: PropTypes.func.isRequired
};
const mapStateToProps = (state) => ({
});
export default connect(mapStateToProps, { fetchDecks })(AddDeck);
