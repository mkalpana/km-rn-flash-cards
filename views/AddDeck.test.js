import React from 'react';
import { AddDeck } from './AddDeck';
import renderer from 'react-test-renderer';

describe('AddDeck', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<AddDeck
      fetchDecks={() => ([])}
    />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
