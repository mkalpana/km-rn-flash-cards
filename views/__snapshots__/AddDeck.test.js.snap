// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDeck renders without crashing 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 30,
        "marginBottom": 10,
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    What is the title of your new deck?
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Deck Title"
    required={true}
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "borderColor": "black",
          "borderRadius": 5,
          "borderWidth": 2,
          "fontSize": 16,
          "marginBottom": 10,
          "padding": 10,
        },
        undefined,
      ]
    }
    value=""
  />
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 40,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "borderColor": "black",
              "borderRadius": 5,
              "borderWidth": 2,
              "padding": 20,
              "paddingLeft": 40,
              "paddingRight": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Create Deck
        </Text>
      </View>
    </View>
  </View>
</View>
`;
